### PR TITLE
Support grouped convolution layer with mkldnn.

### PR DESCRIPTION
**Pull request by chuanqi129**
This PR is for support grouped convolution layer with mkldnn. We have test it based on SE-ResNeXt50 model with this fix, it works fine and the accuracy keep same with claimed. Please have a review on it. @mrysztow @hshen14 @qingqing01 @luotao1 @Superjomn

### CODE REVIEW

File: paddle/fluid/operators/conv_mkldnn_op.cc [hunk 1]
```diff
@@ -303,7 +303,7 @@ class ConvMKLDNNOpKernel : public paddle::framework::OpKernel<T> {
     int groups = ctx.Attr<int>("groups");
 
     // TODO(pzelazko-intel) add support for group convolution and dilation
-    PADDLE_ENFORCE(groups == 1, "group convolution is not implemented yet");
+    // PADDLE_ENFORCE(groups == 1, "group convolution is not implemented yet");
```
Comment (qingqing01) [inline paddle/fluid/operators/conv_mkldnn_op.cc]: Please remove this line.

File: paddle/fluid/operators/conv_mkldnn_op.cc [hunk 2]
```diff
@@ -314,6 +314,19 @@ class ConvMKLDNNOpKernel : public paddle::framework::OpKernel<T> {
     std::vector<int> src_tz = paddle::framework::vectorize2int(input->dims());
     std::vector<int> weights_tz =
         paddle::framework::vectorize2int(filter->dims());
+    int g = std::max(groups, 1);
+    if(g>1){
```
Comment (qingqing01) [inline paddle/fluid/operators/conv_mkldnn_op.cc]: note the code style. please `pip install pre-commit`

File: paddle/fluid/operators/conv_mkldnn_op.cc [hunk 3]
```diff
@@ -340,7 +353,7 @@ class ConvMKLDNNOpKernel : public paddle::framework::OpKernel<T> {
     auto src_md = platform::MKLDNNMemDesc(
         src_tz, platform::MKLDNNGetDataType<T>(), chosen_memory_format);
     auto weights_md = platform::MKLDNNMemDesc(
-        weights_tz, platform::MKLDNNGetDataType<T>(), chosen_memory_format);
+        weights_tz, platform::MKLDNNGetDataType<T>(), (g == 1)?chosen_memory_format:mkldnn::memory::format::goihw);
```
Comment (qingqing01) [inline paddle/fluid/operators/conv_mkldnn_op.cc]: note the code style.

Comment (chuanqi129) [reply paddle/fluid/operators/conv_mkldnn_op.cc]: Thanks, have modified.

**Review by qingqing01 — APPROVED** [review]
LGTM.

**Pull request merged**